ndless-sdk/include/keys.h

```diff
@@ -29,148 +29,147 @@ typedef struct {
-static const t_key KEY_NSPIRE_APOSTROPHE = KEY_(0x1A, 0x001);
-static const t_key KEY_NSPIRE_CAT        = KEYTPAD_(0x1A, 0x002, 0x1A, 0x080);
-static const t_key KEY_NSPIRE_FRAC       = KEYTPAD_(_KEY_DUMMY_ROW, _KEY_DUMMY_COL, 0x1A, 0x100);
-static const t_key KEY_NSPIRE_G          = KEYTPAD_(0x1A, 0x004, 0x16, 0x040);
-static const t_key KEY_NSPIRE_RP         = KEYTPAD_(0x1A, 0x008, 0x1A, 0x008);
-static const t_key KEY_NSPIRE_F          = KEYTPAD_(0x1A, 0x010, 0x18, 0x001);
-static const t_key KEY_NSPIRE_LP         = KEYTPAD_(0x1A, 0x020, 0x1A, 0x020);
-static const t_key KEY_NSPIRE_E          = KEYTPAD_(0x1A, 0x040, 0x18, 0x002);
-static const t_key KEY_NSPIRE_VAR        = KEYTPAD_(0x1A, 0x080, 0x1A, 0x002);
-static const t_key KEY_NSPIRE_D          = KEYTPAD_(0x1A, 0x100, 0x18, 0x004);
-static const t_key KEY_NSPIRE_DEL        = KEYTPAD_(0x1E, 0x100, 0x1A, 0x200);
-static const t_key KEY_NSPIRE_LTHAN      = KEYTPAD_(0x1A, 0x400, _KEY_DUMMY_ROW, _KEY_DUMMY_COL);
-static const t_key KEY_NSPIRE_FLAG       = KEY_(0x1C, 0x001);
-static const t_key KEY_NSPIRE_CLICK      = KEYTPAD_ARROW_(0x1C, 0x002, TPAD_ARROW_CLICK);
-static const t_key KEY_NSPIRE_C          = KEYTPAD_(0x1C, 0x004, 0x18, 0x010);
-static const t_key KEY_NSPIRE_HOME       = KEYTPAD_(0x1C, 0x008, 0x10, 0x200);
-static const t_key KEY_NSPIRE_B          = KEYTPAD_(0x1C, 0x010, 0x18, 0x020);
-static const t_key KEY_NSPIRE_MENU       = KEY_(0x1C, 0x020);
-static const t_key KEY_NSPIRE_A          = KEYTPAD_(0x1C, 0x040, 0x18, 0x040);
-static const t_key KEY_NSPIRE_ESC        = KEY_(0x1C, 0x080);
-static const t_key KEY_NSPIRE_BAR        = KEY_(0x1C, 0x100);
-static const t_key KEY_NSPIRE_TAB        = KEY_(0x1C, 0x200);
-static const t_key KEY_NSPIRE_EQU        = KEYTPAD_(0x1E, 0x400, 0x18, 0x080);
-static const t_key KEY_NSPIRE_UP         = KEYTPAD_ARROW_(0x1E, 0x001, TPAD_ARROW_UP);
-static const t_key KEY_NSPIRE_UPRIGHT    = KEYTPAD_ARROW_(0x1E, 0x002, TPAD_ARROW_UPRIGHT);
-static const t_key KEY_NSPIRE_RIGHT      = KEYTPAD_ARROW_(0x1E, 0x004, TPAD_ARROW_RIGHT);
-static const t_key KEY_NSPIRE_RIGHTDOWN  = KEYTPAD_ARROW_(0x1E, 0x008, TPAD_ARROW_RIGHTDOWN);
-static const t_key KEY_NSPIRE_DOWN       = KEYTPAD_ARROW_(0x1E, 0x010, TPAD_ARROW_DOWN);
-static const t_key KEY_NSPIRE_DOWNLEFT   = KEYTPAD_ARROW_(0x1E, 0x020, TPAD_ARROW_DOWNLEFT);
-static const t_key KEY_NSPIRE_LEFT       = KEYTPAD_ARROW_(0x1E, 0x040, TPAD_ARROW_LEFT);
-static const t_key KEY_NSPIRE_LEFTUP     = KEYTPAD_ARROW_(0x1E, 0x080, TPAD_ARROW_LEFTUP);
-static const t_key KEY_NSPIRE_SHIFT      = KEYTPAD_(0x1A, 0x200, 0x1E, 0x100);
-static const t_key KEY_NSPIRE_CTRL       = KEY_(0x1E, 0x200);
-static const t_key KEY_NSPIRE_DOC        = KEYTPAD_(_KEY_DUMMY_ROW, _KEY_DUMMY_COL, 0x1C, 0x008);
-static const t_key KEY_NSPIRE_TRIG       = KEYTPAD_(_KEY_DUMMY_ROW, _KEY_DUMMY_COL, 0x12, 0x200);
-static const t_key KEY_NSPIRE_SCRATCHPAD = KEYTPAD_(_KEY_DUMMY_ROW, _KEY_DUMMY_COL, 0x1A, 0x400);
+extern const t_key KEY_NSPIRE_RET;
```

Comment:
This means that every use of any of these variables will also pull in all others via keys.o. Can you compare the impact?

Reply:
Obviously this is a mistake on my part... Let me fix that.

Reply:
Actually i fixed it in https://github.com/gameblabla/Ndless/commit/7353060ede8951f118ac210a078790f0590b267c since it wouldn't build ofc but perhaps i should redo my patches and force push ?

Reply:
You can force push at any time - GH reviews work well with that.